02-peer-protocol.md

```diff
@@ -985,6 +1050,41 @@ basepoint for the peer's revocation basepoint.  This allows the peer to
 return channel-assignable errors before the *accepter*'s revocation
 basepoint is known.
 
+If the opener requested a `option_will_fund` peer to contribute funds to
+this channel, via the inclusion of a `will_fund` tlv in their
+`open_channel2`, the peer may or may not choose to accept the offer.
+
+If the accepter node has run out of available funds, they should
+either fail the negotiation or reply with a `funding_satoshi` set
+to 0msat and no `will_fund` TLV, allowing the peer to fail the negotiation.
+
+`funding_weight` is the weight it will cost the node to
+contribute the offered funds to the channel. This is added to the
+total lease fee at a rate of `funding_weight` times the
+`funding_feerate_perkw` established in `open_channel2`.
+
+The total lease fee is added to the accepter node's output in the
+commitment transaction. It is equal to the `funding_fee_base_sat` +
+`accept_channel2`.`funding_satoshis` * `funding_fee_proportional_basis`
++ `funding_feerate_perkw` * `funding_weight` / 1000, rounded
+down to the nearest satoshi. See [Appendix A: The lease fee](#appendix-a-the-lease-fee).
+
+`signature` signs the accepter node's `funding_pubkey`, the opener's
+`locktime`, and the returned `channel_fee` parameters. This provides the opener
+with proof in the case the accepter does not maintain the stated
+`channel_fee` rates during the channel's funding lease.
+
+If the initiating node sent `request_funds` and the accepting node replied
+with `will_fund` this channel is considered 'leased'.  The lease ends
+4032 blocks after the `open_channel2` `blockheight`, i.e.
+`lease_end` is defined as the `open_channel2`.`blockheight` + 4032.
+
+During the lease, the initiator should consider limiting
+the rate funds are routed through the channel as this provides the opener
+the opportunity to move the funds elsewhere (and renders the
+lease fairly useless). A sensible policy here depends on the motivation of the
+opener in acquiring the leased funds.
```

Comment:
This causes a weird case, where Alice requests 1M sats, Bob has 1M+dust sats available.  If Bob puts that dust into the channel, Alice pays a bit extra for it, which is weird.  Is this worth fixing?  Maybe not?  The only fix I can think of is to use min(`accept_channel2`.`funding_satoshis`, `open_channel2.request_funds.requested_sats`) * `funding_fee_proportional_basis` instead of `accept_channel2`.`funding_satoshis` * `funding_fee_proportional_basis` but that's also a bit messy?

Reply:
Min sounds reasonable to me.

Reply:
Yes this is sensible. :+1: